colossalai/utils/memory_tracer/memstats_collector.py

```diff
@@ -31,57 +31,34 @@ def __init__(self) -> None:
         self._sampling_time = []
 
         self._start_flag = False
-        self._period_idx = 0
+        self._step_idx = 0
+        self._step_total = 0
 
-    def overall_mem_stats(self, device_type: str):
+    def overall_mem_stats(self, device_type: str) -> List[int]:
         if device_type == 'cuda':
             return self._overall_cuda_list
         elif device_type == 'cpu':
             return self._overall_cpu_list
         else:
             raise TypeError
 
-    def model_data_list(self, device_type: str, unit: str = 'B') -> List[int]:
-        if unit == 'GB':
-            scale = 1e9
-        elif unit == 'MB':
-            scale = 1e6
-        elif unit == 'KB':
-            scale = 1e3
-        elif unit == 'B':
-            scale = 1
-        else:
-            raise TypeError
-
+    def model_data_list(self, device_type: str) -> List[int]:
         if device_type == 'cuda':
-            return [elem / scale for elem in self._model_data_cuda_list]
+            return self._model_data_cuda_list
         elif device_type == 'cpu':
-            return [elem / scale for elem in self._model_data_cpu_list]
-        else:
-            raise TypeError
-
-    def non_model_data_list(self, device_type: str, unit: str = 'B') -> List[int]:
-        """Non model data stats
-        """
-        if unit == 'GB':
-            scale = 1e9
-        elif unit == 'MB':
-            scale = 1e6
-        elif unit == 'KB':
-            scale = 1e3
-        elif unit == 'B':
-            scale = 1
+            return self._model_data_cpu_list
         else:
             raise TypeError
 
+    def non_model_data_list(self, device_type: str) -> List[int]:
         if device_type == 'cuda':
-            return [elem / scale for elem in self._non_model_data_cuda_list]
+            return self._non_model_data_cuda_list
         elif device_type == 'cpu':
-            return [elem / scale for elem in self._non_model_data_cpu_list]
+            return self._non_model_data_cpu_list
         else:
             raise TypeError
 
-    def max_non_model_data(self, device_type: str) -> int:
+    def next_max_non_model_data(self, device_type: str) -> int:
```

Comment:
next_period_non_model_data?
That is more straitforwards

Reply:
or next_period_non_model_data_usage?

Reply:
Is `future_non_model_data_usage` ok?

Reply:
done